core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -1215,15 +1227,22 @@ class LogManager(logDirs: Seq[File],
             cleaner.updateCheckpoints(removedLog.parentDirFile, partitionToRemove = Option(topicPartition))
           }
         }
-        removedLog.renameDir(UnifiedLog.logDeleteDirName(topicPartition), false)
+        if (isStray) {
+          // Move aside stray partitions, don't delete them
+          removedLog.renameDir(UnifiedLog.logStrayDirName(topicPartition), false)
+          info(s"Log for partition ${removedLog.topicPartition} is renamed to ${removedLog.dir.getAbsolutePath}")
```

Comment:
We should mention that it is a stray log, right? i.e. that it wasn't found in the LeaderAndIsrRequest

Reply:
Yea fair point. It will say "renaming to XXX-stray" but it would better to be obvious about it